lib/stripe/plan.ex

```diff
@@ -14,27 +14,15 @@ defmodule Stripe.Plan do
   Stripe API reference: https://stripe.com/docs/api#plan
   """
 
-  alias Stripe.Util
-
   @type t :: %__MODULE__{}
 
   defstruct [
     :id, :amount, :created, :currency, :interval, :interval_count,
     :livemode, :metadata, :name, :statement_descriptor, :trial_period_days
   ]
 
-  @response_mapping %{
-    id: :string,
-    amount: :integer,
-    created: :datetime,
-    currency: :string,
-    interval: :string,
-    interval_count: :integer,
-    livemode: :boolean,
-    metadata: :metadata,
-    name: :string,
-    statement_descriptor: :string,
-    trial_period_days: :integer
+  @relationships %{
+    created: DateTime
   }
```

Comment:
Hm, I can't think of way to distinguish an integer that is a timestamp from an integer that does represent a timestamp.

Ideally, I'd like to remove the use of `relationships` from the modules as the knowledge of "keys that qualify as structs" isn't something the modules should have to manage. One should not create a new module and be required to define `relationships` on it. Earlier in this PR, I started down the path of removing this mapping entirely but then decided that would be best done in a separate PR. 

I think we could achieve this by creating a `lib/defined_struct_keys.ex` file that is the sole source of which data is mapped to a struct. OR, it could be done by using the `object` value in the Stripe response information in combination with pattern matching functions that target the keys `created, trial_end, trial_start` etc. 

I would recommend the `DefinedStructKeys` approach as I find it more obvious and it's easy maintain.